Bottom: 1a06d70a6c6b7c501a7e22e12b5caad99e75a2e3
Top:    b12b20bfc6b0a7ec8cec4cc6cf3a9083fb2f282e
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-16 19:30:35 +0800

Refresh of macvtap-xdp-fast-rx-support

---

diff --git a/drivers/net/macvtap.c b/drivers/net/macvtap.c
index 24789df..e85abad 100644
--- a/drivers/net/macvtap.c
+++ b/drivers/net/macvtap.c
@@ -1324,11 +1324,19 @@ static int macvtap_recvmsg(struct socket *sock, struct msghdr *m,
 	return ret;
 }
 
+static int macvtap_xdp_peek_len(void *data)
+{
+	struct xdp_buff *buff = data;
+
+	return buff->data_end - buff->data;
+}
+
 static int macvtap_peek_len(struct socket *sock)
 {
 	struct macvtap_queue *q = container_of(sock, struct macvtap_queue,
 					       sock);
-	return skb_array_peek_len(&q->skb_array);
+//	return skb_array_peek_len(&q->skb_array);
+	return PTR_RING_PEEK_CALL(&q->xdp_array, macvtap_xdp_peek_len);
 }
 
 /* Ops structure to mimic raw sockets with tun */
